include/rocksdb/utilities/object_registry.h

```diff
@@ -47,54 +46,137 @@ class ObjectLibrary {
   class Entry {
    public:
     virtual ~Entry() {}
-    Entry(const std::string& name) : name_(std::move(name)) {}
+    virtual bool Matches(const std::string& target) const = 0;
+    virtual const char* Name() const = 0;
+  };
+
+  // Class for matching target strings to a pattern.
```

Comment:
I think it's worth pointing out a key difference from regexes:
* In matching, the left-most match for one piece of a pattern is the only one considered (e.g. if next separator is "::" then "::" cannot be in the current piece), thus
* No backtracking is needed for matching, so matching is reliably efficient

Reply:
Added a comment to PatternEntry.  LMK if you want more and what else you would like to see